### ISSUE TITLE
Compact leaves blank space left between widgets in random situations.

### ISSUE THREAD

**Issue by NoMercy235** [tiberiuzuld/angular-gridster2#246]
Hello!

I have noticed that, in some cases, when the widgets arrange themselves (e.g. coordinates are not specified or after moving widgets that affect the position of many others), you are left with a weird blank space like shown below:

![image](https://user-images.githubusercontent.com/24717857/36251763-09b0b346-124b-11e8-8023-aa10e3465058.png)

If the user tries to move that widget or the screen size changes, it gets to where it was supposed to be. Even switching between browser tabs or simply refreshing the page could lead to the widget rearranging itself. I've tried many times to reproduce this on your demo page, but I never could do it. Do you have any idea where I should look for in order to debug this?

Here's my config:

```
gridsterOptions: GridsterConfig = {
    compactType: 'compactUp&Left',
    gridType: 'fixed',
    maxCols: DashboardComponent.GRIDSTER.maxCols,
    maxRows: DashboardComponent.GRIDSTER.maxRows,
    // fixedColWidth: calculated based on [container's width / max columns]
    fixedRowHeight: DashboardComponent.GRIDSTER.fixedRowHeight,
    maxItemRows: DashboardComponent.GRIDSTER.maxItemRows,
    margin: DashboardComponent.GRIDSTER.margin,
    mobileBreakpoint: DashboardComponent.GRIDSTER.mobileBreakpoint,
    outerMargin: true,
    keepFixedHeightInMobile: true,
    keepFixedWidthInMobile: false,
    pushItems: this.customizeState,
    swap: this.customizeState,
    ignoreMarginInRow: true,
    draggable: {
        enabled: this.customizeState,
        ignoreContent: true,
        dragHandleClass: 'gridster-drag-handle',
    },
    resizable: {
        enabled: this.customizeState,
    }
};
```

**Comment by tiberiuzuld** [comment]
Hi,
I think it has to do with the angular check for dirty or something like that.
At least the symptoms point to that.

Otherwise I have no idea.

**Comment by NoMercy235** [comment]
Yeah, I was also thinking of this, since I frequently use `OnPush` change detection strategy, but even without it, the issue seems to occur at random times. Anyway, I'll let you know if I find anything.

**Comment by tiberiuzuld** [comment]
Hi @NoMercy235 ,
Did you find a way to fix your issue?
Will close this issue since don't think is because of the library.

**Comment by NoMercy235** [comment]
Hello @tiberiuzuld !

I tried your suggestion to see if Angular's the one at fault with its detection strategy, but I wasn't successful in my inquiries. After that, I tried replicating the issue again on your demo, and I think I've found a setup that works and is similar to what I have.

The grid config only has to have this:
Grid type - Horizontal Fixed
Compact Type - Up & Left

Then, try to make a layout like this one:

![layout 2](https://user-images.githubusercontent.com/24717857/36659697-f68d5fd0-1add-11e8-92e2-5343b9324b48.png)

Then try to remove the top-left widget. The key, I think, is to make the widget below the top-left one to be big enough, relative to the ones on its right, to leave space between them after it's removed. This is the result I get after removing it:

![layout 2 broken](https://user-images.githubusercontent.com/24717857/36659777-42b62608-1ade-11e8-96a0-ddcb29fdc227.png)

**Comment by tiberiuzuld** [comment]
Yes was able to reproduce it.
Seems to be a bug from the compact service.
Will try to have it fixed soon.

In the mean time I think if you call the `this.options.api.resize` or `this.options.api.optionsChanged` it will remove the gap.

Thanks

**Comment by NoMercy235** [comment]
*Update*: Calling either `resize` or `optionsChanged` within a `setTimeout()` fixes the issue. It needs the extra tick, otherwise, nothing changes.

**Comment by tiberiuzuld** [comment]
Fixed in 4.7.0

**Comment by NoMercy235** [comment]
Works perfectly! Thank you!
